lib/token-store.js

```diff
@@ -64,6 +65,11 @@ module.exports = function(tokens) {
         range = tokens[index].range;
         starts[range[0]] = index;
         ends[range[1]] = index;
+
+        // allows fast lookup of any source index
+        for (var j = range[0]; j <= range[1]; j++) {
+            ranges[j] = index;
+        }
```

Comment:
is this the main source of the perf hit you're seeing? I'm going to look today or tomorrow into possibly storing tokens in a range tree.